src/dialog-utilities.js

```diff
@@ -0,0 +1,39 @@
+import {Origin} from 'aurelia-metadata';
+
+/**
+* Returns a view model based on a given instruction
+*
+* @param {Object} instruction the instruction from which the view model is determined
+* @return {Promise} the view model
+*/
+export function getViewModel(instruction: any, compositionEngine: CompositionEngine) {
+  if (typeof instruction.viewModel === 'function') {
+    instruction.viewModel = Origin.get(instruction.viewModel).moduleId;
+  } else if (typeof instruction.viewModel === 'string') {
+    return compositionEngine.ensureViewModel(instruction);
+  }
+
+  return Promise.resolve(instruction);
+}
+
+/**
+* Invokes a lifecycle event on a given instance
+*
+* @param {Object} instance  the instance to invoke the event on
+* @param {String} name      the event to invoke
+* @param {Object} [model]   the optional model to invoke the event with
+* @return {Promise} the result of the invocation
+*/
+export function invokeLifecycle(instance: any, name: string, model: any) {
```

Comment:
unless I'm missing something here, I haven't changed the logic of `invokeLifecycle`, I just reordered it to bail early (reduce indentation). The main difference here is the code used to be:
```
if (typeof instance[name] === 'function') {
  /* all of the actual logic with return statements */
}

return Promise.resolve(true); // the default case
```
and was changed to:
```
if (typeof instance[name] !== 'function') {
  return Promise.resolve(true); // bail out early reduce indentation by 1
}

/* all the actual logic */
```